src/mcp-sdk/src/Server/JsonRpcHandler.php

```diff
@@ -54,8 +54,8 @@ public function __construct(
     /**
      * @return iterable<string|null>
      *
-     * @throws ExceptionInterface
-     * @throws \JsonException
+     * @throws ExceptionInterface When a handler throws an exception during message processing
+     * @throws \JsonException     When JSON encoding of the response fails
```

Comment:
```suggestion
     * @throws \JsonException     When JSON decoding of the response fails
```

Reply:
*encoding* looks correct to me, the JsonException is thrown when encoding the response fails (in the encodeResponse method). The method catches and handles `JsonException` from decoding the input gracefully without re-throwing it, so only encoding failures can propagate up 🤔

Reply:
Btw, I dunno if Symfony should try/catch this exception to throw a Symfony Exception instead...

Reply:
I would say this is out of scope of this PR and to be discussed